llvm/tools/objwriter/objwriter.cpp

```diff
@@ -260,6 +265,9 @@ MCSection *ObjectWriter::GetSpecificSection(const char *SectionName,
       // boundaries.
       typeAndAttributes |= MachO::S_ATTR_PURE_INSTRUCTIONS;
     }
+    if (attributes & CustomSectionAttributes_Uninitialized) {
+      typeAndAttributes |= MachO::S_ZEROFILL;
```

Comment:
Is zerofill needed for correctness or just MachO wants that for uninitialized data?

Reply:
Without that we were generating a lot of zeros in the object file I generated.

It does the same thing as `COFF::IMAGE_SCN_CNT_UNINITIALIZED_DATA` for PE and `ELF::SHT_PROGBITS` for ELF below. But yeah, I don't know why LLVM has both these format specific enums _and_ `SectionKind` above.

I haven't actually tried end-to-end on mac, but the object file looks about right now in llvm-objdump.

Reply:
I see. I am not familiar with MachO at this level. I guess it needs to fill the space with zeroes one way or another. So it is either zeroes in the obj file or a fill it with zeroes at load/demand time. The flag could be selecting that.
Makes sense.